ophyd/signal.py

```diff
@@ -95,7 +104,7 @@ def __init__(
         self,
         *,
         name,
-        value=0.0,
+        value=NumericValueInfo(float, (), 0.0),
```

Comment:
Rather than merging the expected info and the default value, can we add a new key(s) of `expected_dtype_and_shape` or `dtype` and `shape`?  In `__init__` it seems reasonable to do a cast, but in `describe` I wonder if we should be checking if the value we have in had is actually the correct dtype/shape or not?

Reply:
In my original proposal I didn't want to add more keyword args, but yeah sure - I introduced `dtype` and `shape` in latest version.

For the check in describe: I didn't change anything yet